packages/core/src/anchor/memory/in-memory-anchor-service.ts

```diff
@@ -310,27 +309,44 @@ export class InMemoryAnchorService implements AnchorService, AnchorValidator {
     }
   }
 
+  async _storeRecord(record: Record<string, unknown>): Promise<CID> {
+    let timeout: any
+
+    const putPromise = this.#dispatcher._ipfs.dag.put(record).finally(() => {
+      clearTimeout(timeout)
+    })
+
+    const timeoutPromise = new Promise((resolve) => {
+      timeout = setTimeout(resolve, 30 * 1000)
+    })
+
+    return await Promise.race([
+      putPromise,
+      timeoutPromise.then(() => {
+        throw new Error(`Timed out storing record in IPFS`)
+      }),
+    ])
+  }
+
   /**
    * Sends anchor commit to Ceramic node and instructs it to publish the commit to pubsub
    * @param streamId
    * @param commit
    */
   async _publishAnchorCommit(streamId: StreamID, commit: AnchorCommit): Promise<CID> {
-    const block = await Block.encode({ value: commit, codec, hasher })
-    const expectedCID = block.cid
-    const stream = await this.#ceramic.applyCommit(streamId, commit, {
-      publish: true,
-      anchor: false,
-    })
-    const commitFound: boolean =
-      null != stream.state.log.find((logEntry) => logEntry.cid.equals(expectedCID))
-    if (!commitFound) {
-      throw new Error(
-        `Anchor commit not found in stream log after being applied to Ceramic node. This most likely means the commit was rejected by Ceramic's conflict resolution. StreamID: ${streamId.toString()}, found tip: ${stream.tip.toString()}`
-      )
+    const anchorCid = await this._storeRecord(commit as any)
+
+    const updateMessage = {
```

Comment:
you can just use `dispatcher.publishTip` for pretty much the whole rest of this function